pageserver/src/http/openapi_spec.yml

```diff
@@ -888,13 +887,27 @@ components:
       type: object
       required:
         - id
+        - attachment_status
       properties:
         id:
           type: string
         current_physical_size:
           type: integer
-        has_in_progress_downloads:
-          type: boolean
```

Comment:
NB: `has_in_progress_downloads` was never part of `required`.

But, IIUC the current console code, at least the one in @NanoBjorn 's PR [Link](https://github.com/neondatabase/cloud/blob/12afcfacf14dd09a3a6c11c7edae85c26f8c25ad/goapp/internal/operationsv2/operation_executor.go#L681-L704
), will exhibit the following behavior :

```go
RETRY:
	for {
		select {
		case <-ctx.Done():
			return ctx.Err()
		case <-timeout.C:
			return fmt.Errorf("tenant attach failed: timeout exceeded")
		case <-retry.C:
			status, err := ps.TenantStatus(ctx, tenantId)
			if err != nil {
				logger.Warn(ctx, fmt.Sprintf("tenant status returned error: %s", err))
				retry.Reset(1)
				continue RETRY
			}

			if status.HasInProgressDownloads != nil && !*status.HasInProgressDownloads {
				break RETRY
			}

			// TODO check lag between safekeepers and pageserver

			retry.Reset(TenantAttachPollSleep)
		}
	}
```

I.e., it will

* observe the field is missing
* re-arm the retry timer (10s)
* retry to poll TenantStatus

So, it will be stuck in that loop, and then whatever happens?

So, I guess academically we'd have to keep the `has_in_progress_downloads` alive, make the console switch to the new `attachment_status` API, then remove the `has_in_progress_downloads` in a separate PR.

**BUT**: Gleb's PR is still WIP and nobody was using the old code that already checked for `HasInProgressDownloads`. So, I think we can get away with just releasing the removal of the old field & addition of the old field at once.

Am I right?

Reply:
> So I guess academically

If the field was part of any autogenerated code which might now start to refuse deserialization we would get problems. Safer to do in phases, the right way. I don't know about the older implementation, but pretty sure there should be autogenerated code. Required and non-required are probably the hardest parts of the spec format, could be either way, or a nil check/assert.

Reply:
> If the field was part of any autogenerated code which might now start to refuse deserialization we would get problems. Safer to do in phases, the right way. 

I checked, the piece of code that uses it is the

```
if status.HasInProgressDownloads != nil && !*status.HasInProgressDownloads {
				break RETRY
			}
```

and that piece of code is not used because the relocation PR is not released in Console.


>  I don't know about the older implementation, but pretty sure there should be autogenerated code. 

Yes but again, the field is marked as optional in OpenAPI spec.
And, the code that is currently in console that checks it, _iff it is present_, is not used right now.

https://github.com/neondatabase/cloud/blob/3b8f74675758c10aa42ebfbf65a4e1883a2ef83d/goapp/internal/operationsv2/operation_executor.go#L636-L638

Reply:
Had conversation with @koivunej earlier today, could convince him this is fine.
Asked control plane team to double-check.